src/main/java/by/custom/utilcalculator/domain/UserProgressStorage.java

```diff
@@ -1,27 +1,52 @@
 package by.custom.utilcalculator.domain;
 
+import org.telegram.telegrambots.meta.api.objects.Message;
+
+import java.io.*;
 import java.util.HashMap;
 
 public class UserProgressStorage {
-    private final HashMap<String, UserProgress> users;
+
+    private static class UserProgressStorageHolder {
+        private static final UserProgressStorage USER_PROGRESS_STORAGE = new UserProgressStorage();
+    }
+
+    private HashMap<String, UserProgress> users;
 
     private UserProgressStorage() {
         users = new HashMap<>();
     }
 
-    public static UserProgressStorage getInstance() {
-        return UserProgressStorageHolder.USER_PROGRESS_STORAGE;
+    public UserProgress getUser(String chatId) {
+        get();
+        return users.get(chatId);
     }
 
-    public UserProgress getUser(String chatId) {
+    public void checkAndCreateNewUser(String chatId) {
         if (!users.containsKey(chatId)) {
-            UserProgress userProgress = new UserProgress();
+            UserProgress userProgress = new UserProgress(chatId);
             users.put(chatId, userProgress);
-            return userProgress;
-        } else return users.get(chatId);
+        }
+        save();
     }
 
-    private static class UserProgressStorageHolder {
-        private static final UserProgressStorage USER_PROGRESS_STORAGE = new UserProgressStorage();
+    public void save() {
```

Comment:
It's strange that `save` do not accept any params. What does it save? It's not clear.
```suggestion
    public void save(UserProgress userProgress) {
```

Reply:
Fixed.